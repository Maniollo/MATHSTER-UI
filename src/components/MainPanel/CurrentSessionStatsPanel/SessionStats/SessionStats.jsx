import React from "react";
import SessionTimer from "./SessionTimer/SessionTimer";

const getPercentageOfCorrect = (noOfCorrect, noOfAll) => noOfCorrect ? Math.floor(noOfCorrect / noOfAll * 100) : 0;

export const SessionStats = ({noOfAll, noOfCorrect, noOfIncorrect}) => <div>
    <span style={{"margin": 10}}>Number of attempts: {noOfAll}</span>
    <span style={{"margin": 10}}>{noOfCorrect} / {noOfIncorrect}</span>
    <span>{getPercentageOfCorrect(noOfCorrect, noOfAll)}%</span>
    <SessionTimer />
</div>
