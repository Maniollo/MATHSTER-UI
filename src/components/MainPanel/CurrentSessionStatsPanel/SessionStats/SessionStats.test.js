import React from "react";
import renderer from 'react-test-renderer';
import {SessionStats} from "./SessionStats";

test('renders correctly', () => {
    const panel = renderer.create(<SessionStats noOfAll={24} noOfCorrect={6} noOfIncorrect={18}/>).toJSON();
    expect(panel).toMatchSnapshot();
})