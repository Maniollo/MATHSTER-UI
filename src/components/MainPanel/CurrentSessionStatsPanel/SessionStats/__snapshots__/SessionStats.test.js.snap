exports[`test renders correctly 1`] = `
<div>
  <span
    style={
      Object {
        "margin": 10,
      }
    }>
    Number of attempts: 
    24
  </span>
  <span
    style={
      Object {
        "margin": 10,
      }
    }>
    6
     / 
    18
  </span>
  <span>
    25
    %
  </span>
  <div>
    <span>
      Session time: 
      00
      :
      00
      :
      00
    </span>
  </div>
</div>
`;
